modules/apps/portal-odata/portal-odata-impl/src/test/java/com/liferay/portal/odata/internal/filter/FilterParserImplTest.java

```diff
@@ -77,6 +77,51 @@ public void testParseNonexistingField() {
 			filterString);
 	}
 
+	@Test
+	public void testParseSuboperationWithDateAndDuration()
+		throws ExpressionVisitException {
+
+		Expression expression = _filterParserImpl.parse(
+			"dateTimeExternal ge (2022-12-28T23:00:00.000Z sub duration'PT24H')");
+
+		Assert.assertNotNull(expression);
+
+		BinaryExpression binaryExpression1 = (BinaryExpression)expression;
```

Comment:
You can't use binaryExpression1, if there is not a binaryExpression2

Reply:
The test has another var called binaryExpression2.